0021-hashed-timelock-agreements/0021-hashed-timelock-agreements.md

```diff
@@ -0,0 +1,139 @@
+
+The functionality necessary for simple payment channels is present in nearly all major blockchains today, including Bitcoin (even without [SegWit](https://en.bitcoin.it/wiki/Segregated_Witness)), Ethereum, XRP, Zcash, and Chain.
+
+### Trustlines
+
+* **Ledger Features Required:** Transfers Between Accounts
+* **Non-Functional Ledger Requirements:** N/A
+* **Money at Risk:** Total Trustline Balance
+* **Examples:** [`ilp-plugin-virtual`](https://github.com/interledgerjs/ilp-plugin-virtual)
+
+When a ledger does not provide any support for Interledger, parties can still connect to the Interledger using trustlines. The sender prepares the transfer by sending the recipient a message including the hashlock and timeout. If the recipient produces the hash preimage before the timeout, the sender's debt is increased by the transfer amount. Like with simple payment channels, the participants must agree on how to solve disputes, including disagreements about whether a hash preimage was submitted in time. The sender can continue sending payments until they reach the maximum balance allowed by the recipient. At that point, the sender settles their balance by making a transfer on the ledger.
+
+In this model, one of the parties must trust the other for the full outstanding balance of the trustline. This may be suitable in cases where the sender and recipient are friends or have a long-standing business relationship.
+
+Notably, trustlines work over any type of "ledger" -- including cash! As long as there is a way to transfer funds between the sender and recipient, they can send Interledger payments and settle only when the trust limit is reached. If the payment flows in both directions are balanced, there may never even be a need to settle.
+
+## Conclusion: Integrating All the Ledgers
+
+Interledger provides secure multi-hop payments over any type of ledger by using bilateral Hashed-Timelock Agreements (HTLAs). This works with public and private blockchains, centralized ledgers, payment channels, and even with cash or cases where there is no ledger. There is a spectrum of HTLA types that require varying levels of ledger functionality and bilateral trust. However, because Interledger isolates participants from risk, these options and tradeoffs are only the concern of individual participants and their direct connections. No one is affected by the choices of indirect connections, so all are free to select their ledgers and connectors and use the type of agreement that suits them.
+
+_Internet "hat tip": the importance of being able to integrate **any** type of network was demonstrated by [RFC 1149: A Standard for the Transmission of IP Datagrams on Avian Carriers](https://tools.ietf.org/html/rfc1149)._
+
+## Appendix: Additional HTLA Types
+
+This appendix includes additional HTLA types that are possible but were not mentioned above either because they are less desirable or have yet to be implemented.
+
+### Third Party Escrow
```

Comment:
What's the difference between using a third party escrow provider and just peering with a connector you trust? If there is a difference, it might be worth explaining how the risk is different.